### ISSUE TITLE
Last.fm scrobbling

### ISSUE THREAD

**Issue by YodaEmbedding** [jpochyla/psst#424]
The official Spotify clients and `spotifyd` will scrobble currently playing tracks appropriately using [the official last.fm / Spotify service](https://support.last.fm/t/spotify-scrobbling/189). Possibly, implementing the "Connect" feature tracked in https://github.com/jpochyla/psst/issues/33 will automatically implement this feature as a side-effect, as [this suggests](https://github.com/dmfutcher/spotify-connect-scrobbler#spotify-connect-scrobbler):

> ...Spotify changed the way their Last.fm integration works. Scrobbling now happens on Spotify's servers, rather than the client, so all Connect devices can now scrobble.

Workarounds: using a local MPRIS based scrobbler, such as [`scrobblez`](https://github.com/YodaEmbedding/scrobblez). EDIT: blocked by https://github.com/jpochyla/psst/issues/368

**Comment by kingosticks** [comment]
I don't believe you get any scrobbling by implementing Spotify Connect. I don't think you get it by implementing the reporting/event-service stuff either (see https://github.com/librespot-org/librespot/issues/913, https://github.com/librespot-org/librespot/discussions/626), I thought each client-side scrobbled directly, but I might be wrong about that.

**Comment by marcfusch** [comment]
>  Workarounds: using a local MPRIS based scrobbler, such as [scrobblez](https://github.com/YodaEmbedding/scrobblez).

Such scrobblers do not work on MacOS. So implementing one to psst would be a good idea.

**Comment by mrkkr** [comment]
any updates about this?

**Comment by jacksongoode** [comment]
We are happy to accept PR's! And I'd be more than happy to collab if someone gets one started :)

**Comment by mrkkr** [comment]
> no unfortunatly :/

I thought that Spotify is natively integrated with last.fm when user provides credentials.

is there any 3rd party solution to workaround this?

**Comment by kingosticks** [comment]
There's https://github.com/Ladbaby/lastfm-scrobbler but requires the media player to have MPRIS support, does Psst have that?

**Comment by marcfusch** [comment]
Hi, I'm currently working on a project and I want to implement a last.fm scrobbler (or other scrobblers) to psst since its a feature I use a lot on the og client. I want to do this locally because writing the part for the client to send tracks to spotify servers is pretty hard (needs spotify connect).
I already did some modifications localy to add a special tab in the preferences menu and to grab the track name + artists.
Do you, regular contributors to the project, have any advice for me to not mess too much with the codebase?
Thanks.

PS: This is a copy of an already existing (but newer) issue #590

**Comment by jacksongoode** [comment]
Hey @marcfusch I feel like this would be a great first PR and I would happily advise and contribute!

**Comment by kingosticks** [comment]
Extra points if you can make it compatible with libre.fm also!

**Comment by marcfusch** [comment]
I will first try to get lastfm working,then i will propably add listenbrainz / librefm, I just need to figure out how authentication works on the library i am using.

**Comment by SO9010** [comment]
Ok so I had a little look in for you @marcfusch. Ok so i think it bakes sense to add a new file in psst-gui/src/webapi/ called scrobblers.rs where you have all of the api requests and set up, you can probably take inspiration from the client.rs file. Then you want to call it in the playback controller and then update in a similar way to how we update the lyrics view :) : 

``` rust 
    fn update_lyrics(&self, ctx: &mut EventCtx, data: &AppState, now_playing: &NowPlaying) {
        if matches!(data.nav, Nav::Lyrics) {
            ctx.submit_command(lyrics::SHOW_LYRICS.with(now_playing.clone()));
        }
    }

    // fn update_scrobler(&self, now_playing: &NowPlaying) {
    //     if let Some(scrobler) = &data.scrobler {
    //         scrobler.update_now_playing(data.playback.now_playing.as_ref());
    //     }
    // }
`` `

**Comment by SO9010** [comment]
Or it could make more sense to add this into psst core if you are able to use the song id, in that case you could make a new file again and then use id.to_base16()

**Comment by SO9010** [comment]
Closed with  #594
